### PR TITLE
Add support for linking openssl dynamically

### PR DESCRIPTION
**Pull request by simo5**
For non-FIPS builds we can link to the system libcrypto without issues, so add that as an option.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -136,7 +137,34 @@ fn build_ossl() {
         .expect("Couldn't write bindings!");
 }
 
+#[cfg(feature = "dynamic")]
```
Comment (Jakuje) [inline build.rs:140]: should we do the CI target with dynamic linking to make sure this keeps working?

File: build.rs [hunk 2]
```diff
@@ -136,7 +137,34 @@ fn build_ossl() {
...
+        .allowlist_item("ossl_.*")
+        .allowlist_item("OSSL_.*")
+        .allowlist_item("openssl_.*")
+        .allowlist_item("OPENSSL_.*")
+        .allowlist_item("CRYPTO_.*")
+        .allowlist_item("c_.*")
+        .allowlist_item("EVP_.*")
+        .allowlist_item("evp_.*")
+        .allowlist_item("BN_.*")
+        .allowlist_item("LN_aes.*")
```
Comment (Jakuje) [inline build.rs:157]: Can this list be somehow reused from the non-dynamic linking as it is exactly same and keeping them in sync is quite error-prone?

File: build.rs (hunk 1 above)
Comment (simo5) [reply build.rs:140]: good idea!

File: build.rs (hunk 2 above)
Comment (simo5) [reply build.rs:157]: Yeah, let me add an auxiliary function for that

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -94,6 +97,9 @@ jobs:
           if [ "${{ matrix.name }}" = "release" ]; then
             cargo test --release
           fi
+          if [ "${{ matrix.name }}" = "dynamic" ]; then
+            cargo test --features dynamic
```
Comment (Jakuje) [inline .github/workflows/build.yml:101]: maybe one more idea that can be put in place later in follow-up PR or in long term is to run this on different OS/versions, not just Fedora as this is no longer staticaly compiled from "known-to-be-good" openssl version.

**Review by Jakuje — APPROVED** [review]

**Pull request merged**